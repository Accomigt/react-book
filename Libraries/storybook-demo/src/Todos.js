import React from 'react';
import styled from 'styled-components';

const Todo = styled.div`
  box-shadow: 0 0 5px grey;
  padding: 10px;
  margin-bottom: 10px;
`;

const Todos = ({ todos }) => (
  <React.Fragment>
    <h3>List of todos</h3>
    {todos.map(t => <Todo key={t.title}>{t.title}</Todo>)}
  </React.Fragment>
);

export default Todos;
