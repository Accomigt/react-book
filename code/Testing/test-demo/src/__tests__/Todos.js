import renderer from 'react-test-renderer';
import React from 'react';

import Todos from '../Todos';

test('Todo - should create snapshot', () => {
  const component = renderer.create(
    <Todos todos={[{ title: 'item1', description: 'an item'}, {title: 'item2', description: 'another item'}]} />,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
})
