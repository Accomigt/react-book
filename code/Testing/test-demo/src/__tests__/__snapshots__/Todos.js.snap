// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo - should create snapshot 1`] = `
Array [
  <h3
    className=""
    data-testid="item"
>
    item1
</h3>,
  <div>
    an item
</div>,
  <button
    onClick={[Function]}
>
    Select
</button>,
  <h3
    className=""
    data-testid="item"
>
    item2
</h3>,
  <div>
    another item
</div>,
  <button
    onClick={[Function]}
>
    Select
</button>,
]
`;
